### PR TITLE
[FIX] l10n_es_pos: not invoice when settling customer accounts

### PR DESCRIPTION
**Pull request by sbel-odoo**
Currently when settling a customer account, we try to create a simplified invoice which is nt possible since we do not have products in the order. Users using the Spanish localization cannot settle customer accounts.

Steps to reproduce:
-------------------
* Install **l10n_es_pos** and switch to the ES Company
* Open shop session
* Add products to the order, select any customer, pay with customer account
* Select **New order**
* Select the previous customer and select **Settle due accounts**
* Select any payment method
* Validate order -> Yes
> Observation: Cannot invoice empty order

Why the fix:
------------
Spain requires to invoice all orders. In Pos, everything is considered as an order but settling customer accounts does not fall under the definition of an order outside of Odoo (no transfer of product, amount=0).

Therefore when we settle customer account we skip the automatic invoicing and fallback on the pos bahavior without the localization.

opw-4185144

Enterprise PR: https://github.com/odoo/enterprise/pull/71958

### CODE REVIEW

**Comment by sbel-odoo** [comment]
@robodoo up to saas-17.4

File: addons/l10n_es_pos/tests/test_frontend.py [hunk 1]
```diff
@@ -41,6 +41,58 @@ def get_number_of_regular_invoices():
...
+
+        # create customer account payment method
+        self.customer_account_payment_method = self.env['pos.payment.method'].create({
+            'name': 'Customer Account',
+            'split_transactions': True,
+        })
+        # add customer account payment method to pos config
+        self.main_pos_config.write({
+            'payment_method_ids': [(4, self.customer_account_payment_method.id, 0)],
+        })
```
Comment (clbr-odoo) [inline addons/l10n_es_pos/tests/test_frontend.py:56]: Please either use Command or just this
`self.main_pos_config.payment_method_ids = self.customer_account_payment_method.id`

**Review by clbr-odoo — CHANGES_REQUESTED** [review]
Small nitpicking :slightly_smiling_face: 

From accounting point of view it makes sense to me to not invoice the settling of customer accounts, and @sbel-odoo confirmed with CHKL.

@caburj Already approved. 
For the record imo this could/should be handled in generic and not only in Spanish POS, but it's not easily doable in stable and seems to work already in 18.0. Maybe worth to double check when you're back JCB :slightly_smiling_face: 
(So many (nice) changes in 18.0 POS that its hard to follow :p )

**Review by clbr-odoo — APPROVED** [review]

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
robodoo r+
